### PR TITLE
Simple approach to connecting failing transactions properly

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #979** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/979?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/979?utm_source=stack-comment).

File: lib/block_view.go [hunk 1]
```diff
@@ -4082,10 +4086,27 @@ func (bav *UtxoView) ConnectBlock(
 				return nil, errors.Wrapf(err, "ConnectBlock: txn #%d should not connect based on "+
 					"TxnConnectStatusByIndex but err is nil", txIndex)
 			}
```
Comment (tholonious) [inline lib/block_view.go:4088]: I think you want to remove this `if err == nil` check from here. We haven't tried to connect hte txn yet, and so shouldn't check for any errors on it

**Review by tholonious — APPROVED** [review]

File: lib/block_view.go [hunk 2]
```diff
@@ -4058,35 +4058,52 @@ func (bav *UtxoView) ConnectBlock(
...
-		// connecting blocks. We skip this check because computing the transaction's size
-		// would slow down block processing significantly. We should figure out a way to
-		// enforce this check in the future, but for now the only attack vector is one in
-		// which a miner is trying to spam the network, which should generally never happen.
-		utxoOpsForTxn, totalInput, totalOutput, currentFees, err := bav.ConnectTransaction(txn, txHash, 0, uint32(blockHeader.Height), blockHeader.TstampNanoSecs, verifySignatures, false)
-		_, _ = totalInput, totalOutput // A bit surprising we don't use these
 		// After the PoS cutover, we need to check if the transaction is a failing transaction.
 		txnConnects := blockHeight < uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) ||
 			(txIndex == 0 && txn.TxnMeta.GetTxnType() == TxnTypeBlockReward) ||
 			desoBlock.TxnConnectStatusByIndex.Get(txIndex-1)
```
Comment (AeonSw4n) [inline lib/block_view.go:4064]: What happens if `txIndex = 0 && txn.TxnMeta.GetTxnType() != TxnTypeBlockReward`? Won't desoBlock.TxnConnectStatusByIndex.Get(-1) panic?

Comment (lazynina) [reply lib/block_view.go:4064]: On line 4038 in block_view.go, we verify that the first transaction is a block reward transaction.

Comment (AeonSw4n) [reply lib/block_view.go:4064]: Ah, good, good!

**Pull request merged**